Source/NSSplitViewItem.m

```diff
@@ -0,0 +1,177 @@
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArchiver.h>
+#import "AppKit/NSSplitViewItem.h"
+
+@implementation NSSplitViewItem
+- (instancetype) initWithViewController: (NSViewController *)viewController
+{
+  self = [super init];
+  if (self != nil)
+    {
+      ASSIGN(_viewController, viewController);
+    }
+  return self;
+}
+
++ (instancetype) contentListWithViewController: (NSViewController *)viewController
+{
+  return [[NSSplitViewItem alloc] initWithViewController: viewController];
+}
+
++ (instancetype) sidebarWithViewController: (NSViewController *)viewController
+{
+  return [[NSSplitViewItem alloc] initWithViewController: viewController];
+}
+
++ (instancetype) splitViewItemWithViewController: (NSViewController *)viewController
+{
+  return [[NSSplitViewItem alloc] initWithViewController: viewController];  
+}
+
+- (CGFloat) automaticMaximumThickness
+{
+  return _automaticMaximumThickness;
+}
+
+- (void) setAutomaticMaximumThickness: (CGFloat)f
+{
+  _automaticMaximumThickness = f;
+}
+
+- (CGFloat) preferredThicknessFraction
+{
+  return _preferredThicknessFraction;
+}
+
+- (void) setPreferredThicknessFraction: (CGFloat)f
+{
+  _preferredThicknessFraction = f;
+}
+
+- (CGFloat) minimumThickness
+{
+  return _minimumThickness;
+}
+
+- (void) setMinimumThickness: (CGFloat)f
+{
+  _minimumThickness = f;
+}
+
+- (CGFloat) maximumThickness
+{
+  return _maximumThickness;
+}
+
+- (void) setMaximumThickness: (CGFloat)f
+{
+  _maximumThickness = f;
+}
+
+- (/* NSLayoutPriority */ CGFloat) holdingPriority
+{
+  return _holdingPriority;
+}
+
+- (void) setHoldingPriority: (/*NSLayoutPriority*/ CGFloat)hp
+{
+  _holdingPriority = hp;
+}
+
+- (BOOL) canCollapse
+{
+  return _canCollapse;
+}
+
+- (NSSplitViewItemCollapseBehavior) collapseBehavior
+{
+  return _collapseBehavior;
+}
+
+- (BOOL) isSpringLoaded
+{
+  return _springLoaded;
+}
+
+- (void) setSpringLoaded: (BOOL)flag
+{
+  _springLoaded = flag;
+}
+
+- (BOOL) allowsFullHeightLayout
+{
+  return _allowsFullHeightLayout;
+}
+
+- (void) setAllowsFullHeightLayout: (BOOL)flag
+{
+  _allowsFullHeightLayout = flag;
+}
+
+- (NSTitlebarSeparatorStyle) titlebarSeparatorStyle
+{
+  return _titlebarSeparatorStyle;
+}
+
+- (void) setTitlebarSeparatorStyle: (NSTitlebarSeparatorStyle)style
+{
+  _titlebarSeparatorStyle = style;
+}
+
+- (NSViewController *) viewController
+{
+  return _viewController;
+}
+
+- (void) setViewController: (NSViewController *)vc
+{
+  _viewController = vc;
+}
+
+// NSCoding
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSSplitViewItemViewController"])
+        {
+          _viewController = [coder decodeObjectForKey: @"NSSplitViewItemViewController"];
```

Comment:
What about the other instance variables?

Reply:
Working on it.

Reply:
Where?